src/annotator/components/ContentInfoBanner.js

```diff
@@ -8,39 +15,108 @@ import { LabeledButton, Link } from '@hypothesis/frontend-shared';
-          <span>
-            {' '}
-            from <i>{info.item.containerTitle}</i>
-          </span>
-        )}
-        {info.links.nextItem && (
-          <Link href={info.links.nextItem} target="_blank">
-            Next article
-          </Link>
+      </div>
+      <div
+        className={classnames(
+          // Container title (this element) is not shown on narrow screens
+          'hidden',
+          '2xl:block 2xl:whitespace-nowrap 2xl:overflow-hidden 2xl:text-ellipsis'
         )}
+        data-testid="content-container-info"
+      >
+        <span className="font-semibold" title={info.item.containerTitle}>
+          {info.item.containerTitle}
+        </span>
       </div>
-      <div className="text-annotator-base">
-        <LabeledButton onClick={onClose} data-testid="close-button">
-          Close
-        </LabeledButton>
+      <div
+        className={classnames(
+          // Flex layout for item title, next and previous links
+          'flex justify-center items-center gap-x-2'
+        )}
+        data-testid="content-item-info"
+      >
+        <div
+          className={classnames(
+            // Narrower viewports center this flex content:
+            // this element is not needed for alignment
+            'hidden',
+            // Wider viewports align this flex content to the right:
+            // This empty element is needed to fill extra space at left
+            '2xl:block 2xl:grow'
+          )}
+        />
+        {info.links?.previousItem && (
+          <>
+            <Link
+              classes="flex gap-x-1 items-center text-annotator-sm whitespace-nowrap"
+              title={`Open previous item in a new window`}
+              href={info.links.previousItem}
+              underline="always"
+              data-testid="content-previous-link"
+            >
+              <CaretLeftIcon className="w-em h-em" />
+              <span>Previous</span>
+            </Link>
+            <div className="text-annotator-sm">|</div>
+          </>
+        )}
+        <div
+          className={classnames(
+            // This element will shrink and truncate fluidly.
+            // Overriding min-width `auto` prevents the content from overflowing
+            // See https://stackoverflow.com/a/66689926/434243.
+            'min-w-0 whitespace-nowrap overflow-hidden text-ellipsis shrink font-medium'
+          )}
+        >
+          <LinkUnstyled
+            title={`Open "${info.item.title}" in a new window`}
+            href="http://www.example.org"
+            data-testid="content-item-link"
+          >
+            {info.item.title}
+          </LinkUnstyled>
+        </div>
+
+        {info.links?.nextItem && (
+          <>
+            <div className="text-annotator-sm">|</div>
+            <Link
+              title={`Open next item in a new window`}
+              classes="flex gap-x-1 items-center text-annotator-sm whitespace-nowrap"
+              href="https://jstor.org"
```

Comment:
The `href` here should be `info.links.nextItem`.

Reply:
Yes, it should. Boy, I'm really on fire for quality control today, eh? 😬 Thanks for your patience.